openupgrade_scripts/scripts/stock/17.0.1.1/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,142 @@
+stock        / stock.picking            / immediate_transfer (boolean)  : DEL
+stock        / stock.picking            / message_main_attachment_id (many2one): DEL relation: ir.attachment
+stock        / stock.picking            / move_ids_without_package (one2many): is now stored
+stock        / stock.picking            / move_ids_without_package (one2many): not a function anymore
+stock        / stock.picking            / move_line_nosuggest_ids (one2many): DEL relation: stock.move.line
+stock        / stock.picking            / picking_properties (properties): NEW hasdefault: compute
+stock        / stock.picking            / rating_ids (one2many)         : NEW relation: rating.rating
+stock        / stock.picking            / return_id (many2one)          : NEW relation: stock.picking
+stock        / stock.picking            / return_ids (one2many)         : NEW relation: stock.picking
+stock        / stock.picking            / show_operations (boolean)     : not a function anymore
+stock        / stock.picking            / show_operations (boolean)     : now related
+
+stock        / stock.picking.type       / auto_print_delivery_slip (boolean): NEW
+stock        / stock.picking.type       / auto_print_lot_labels (boolean): NEW
+stock        / stock.picking.type       / auto_print_package_label (boolean): NEW
+stock        / stock.picking.type       / auto_print_packages (boolean) : NEW
+stock        / stock.picking.type       / auto_print_product_labels (boolean): NEW
+stock        / stock.picking.type       / auto_print_reception_report (boolean): NEW
+stock        / stock.picking.type       / auto_print_reception_report_labels (boolean): NEW
+stock        / stock.picking.type       / auto_print_return_slip (boolean): NEW
+stock        / stock.picking.type       / default_location_return_id (many2one): NEW relation: stock.location
+stock        / stock.picking.type       / lot_label_format (selection)  : NEW selection_keys: ['4x12_lots', '4x12_units', 'zpl_lots', 'zpl_units'], hasdefault: default
+stock        / stock.picking.type       / package_label_to_print (selection): NEW selection_keys: ['pdf', 'zpl'], hasdefault: default
+stock        / stock.picking.type       / picking_properties_definition (properties_definition): NEW
+stock        / stock.picking.type       / product_label_format (selection): NEW selection_keys: ['2x7xprice', '4x12', '4x12xprice', '4x7xprice', 'dymo', 'zpl', 'zplxprice'], hasdefault: default
+stock        / stock.picking.type       / show_reserved (boolean)       : now a function
+# NOTHING TO DO: fields in new model
+
+stock        / stock.scrap              / message_main_attachment_id (many2one): DEL relation: ir.attachment
+stock        / stock.scrap              / move_id (many2one)            : DEL relation: stock.move
+stock        / stock.scrap              / move_ids (one2many)           : NEW relation: stock.move
```

Comment:
I think it is necessary to deal with the structural change here. Something like this.
```python
def _update_stock_move_scrap(env):
    sql = 'SELECT id,move_id FROM stock_scrap WHERE move_id IS NOT NULL;'
    env.cr.execute(sql)
    scrap_dict = env.cr.dictfetchall()
    scrap_dict = dict([(item['id'], item['move_id']) for item in scrap_dict])
    for scrap in env['stock.scrap'].search([('id', 'in', list(scrap_dict.keys()))]):
        scrap.write({'move_ids': [(4, scrap_dict[scrap.id])]})
```